src/main/java/androidx/core/widget/ImageView.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.core.widget
+
+import android.graphics.ColorMatrix
+import android.graphics.ColorMatrixColorFilter
+import android.graphics.Paint
+import android.widget.ImageView
+
+/**
+ * Filtering an image based on saturation
+ *
+ * @param saturation Value to set the saturation
+ */
+fun ImageView.filter(saturation: Float) {
+    val colorMatrix = ColorMatrix().apply { setSaturation(saturation) }
+    val paint = Paint().apply { colorFilter = ColorMatrixColorFilter(colorMatrix) }
+    this.setLayerType(ImageView.LAYER_TYPE_HARDWARE, paint)
```

Comment:
Why force a hardware layer? It's wasteful on an ImageView, even with a color matrix filter.

Reply:
When I'm not forcing this to hardware layer, sometimes on several device doesn't work. Do you have any suggestion ?

Reply:
Hard to tell without knowing how you were trying to apply the filter to your image.